src/approval/index.ts

```diff
@@ -0,0 +1,144 @@
+import { S3 } from 'aws-sdk';
+import { Md5 } from 'ts-md5/dist/md5';
+import * as fs from 'fs';
+import * as cli from 'cli-color';
+import * as path from 'path';
+import * as jsdiff from 'diff';
+
+import { Arguments } from 'yargs';
+import { loadStackArgs, parseS3HttpUrl } from '../cfn/index';
+import configureAWS from '../configureAWS';
+
+export async function requestApproveTemplate(argv: Arguments): Promise<number> {
+  const stackArgs = await loadStackArgs(argv);
+  const s3 = new S3();
+
+  await configureAWS(stackArgs.Profile, stackArgs.Region);
+
+  const cfnTemplate = await fs.readFileSync(stackArgs.Template);
+  const cfnTemplateFileName = stackArgs.Template.match(/[a-zA-Z-_0-9]*\.yaml/)![0];
+  const s3Url = parseS3HttpUrl(stackArgs.ApprovedTemplateLocation);
+
+  let hashedKey = new Md5().appendStr(cfnTemplate.toString()).end().toString();
+
+  if (s3Url.key.length > 0) {
+    hashedKey = `${s3Url.key}/${hashedKey}`;
+  }
+
+  try {
+    await s3.headObject({
+      Bucket: s3Url.bucket,
+      Key: `${hashedKey}.yaml`
+    }).promise();
+
+    logSuccess(`👍 Your template has already been approved`);
+  } catch (e) {
+    if (e.code === "NotFound") {
+      await s3.putObject({
+        Body: cfnTemplate,
+        Bucket: s3Url.bucket,
+        Key: `${hashedKey}.yaml.pending`
+      }).promise();
+
+      logSuccess(`Successfully uploaded ${cfnTemplateFileName} to ${stackArgs.ApprovedTemplateLocation}`);
+      logSuccess(`Approve template with: `);
+      console.log(`  iidy approve-template https://s3.amazonaws.com/${s3Url.bucket}/${hashedKey}.yaml.pending`);
+    } else {
+      throw new Error(e);
+    }
+  }
+
+  return 0;
+}
+
+export async function approveTemplate(argv: Arguments): Promise<number> {
+  await configureAWS(argv.profile, "us-east-1");
+  const s3 = new S3();
+  const s3Url = parseS3HttpUrl(argv.filename);
```

Comment:
The same as https://github.com/unbounce/iidy/pull/41#discussion_r160296403, this is a misuse of `parseS3HttpUrl`